sql/src/main/java/com/rilldata/calcite/validators/S3ConnectorValidator.java

```diff
@@ -0,0 +1,42 @@
+package com.rilldata.calcite.validators;
+
+import org.apache.calcite.tools.ValidationException;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class S3ConnectorValidator
+{
```

Comment:
- I was thinking adding validation here can save the unnecessary protobuf serialization but probably its not much savings. Probably runtime might be a better place to do this so that concerns of sql lib are just parsing and AST generation. I do not see any disadvantage in it. Either ways I am fine.  
- Probably we can have enum for possible connectors with required and optional fields, at boot time application can make a call to runtime to get this in json format.

Reply:
runtime can also do fancy validations like checking required configs loaded through env files etc.